pytrip/res/concave_tool.py

```diff
@@ -0,0 +1,306 @@
+import math
+
+# This file contains methods that create from a list of intersections (list of points).
+# Algorithm steps:
+#     1. Divide passed list of intersections into separate groups, that each represents a valid contour.
+#     2. For each group create a contour.
+#         2.1. Divide group of points into parts of contour, polygon representing each part imitates plot of a function
+#         2.2. Connect parts of contour (taking into account direction of connection)
+#     3. Return created contours.
+# Basic heuristics are used:
+#     1. searching for minimal distance between a part of contour and a point and ensuring that relation is symmetrical
+#         (if point A is the closest one to part B, check if part B is the closest one to point A)
+#     2. limiting distance between parts and points based on previous distances
+#         (next point cannot be too far away from parts)
+
+
+class ListEntry:
+    """Special type to hold data and other useful information"""
+    def __init__(self):
+        self.data = []
+        self.last_distance = float('inf')
+
+
+class SpecialPoint:
+    """Special type to hold data and other useful information"""
+    def __init__(self, point):
+        self.point = point
+        self.is_appended = False
+
+
+# -------------------- start block of utility methods --------------------
+magic_max_distance_multiplier = 2
+
+
+def map_points_to_special_points(points):
+    return [SpecialPoint(p) for p in points]
+
+
+def calculate_distance(p, q):
+    """Return value of distance between two points in euclidean 3D space"""
+    return math.sqrt((p[0] - q[0])**2 + (p[1] - q[1])**2 + (p[2] - q[2])**2)
+
+
+def search_for_closest_part(s_point, parts_of_contour):
+    closest_part = None
+    closest_distance = float('inf')
+    for part in parts_of_contour:
+        distance = calculate_distance(s_point.point, part.data[-1])
+        if distance < closest_distance and distance < magic_max_distance_multiplier * part.last_distance:
+            closest_distance = distance
+            closest_part = part
+    return closest_part
+
+
+def search_for_closest_point(part_of_contour, s_points):

```

Comment:
Rename to `closest_point` ?

Reply:
![image](https://user-images.githubusercontent.com/44035398/143677604-b6b31f76-728d-4c48-be10-108c8ec6d943.png)
Then I would have to change variables, which are self-explanatory right now